Services/Diagnostics/LoggingConfig.cs

```diff
@@ -12,39 +12,48 @@ public enum LogLevel
         Error = 40,
         Always = 1000
     }
-    
+
     public interface ILoggingConfig
     {
         LogLevel LogLevel { get; }
         bool LogProcessId { get; }
         bool ExtraDiagnostics { get; }
         string ExtraDiagnosticsPath { get; }
         string DateFormat { get; }
+        string BwListPrefix { get; }
         HashSet<string> BlackList { get; }
         HashSet<string> WhiteList { get; }
+        int DebuggingLogBufferLength { get; }
     }
-    
+
+    // Note: singleton class
     public class LoggingConfig : ILoggingConfig
     {
         public const LogLevel DEFAULT_LOGLEVEL = LogLevel.Warn;
         public const string DEFAULT_DATE_FORMAT = "yyyy-MM-dd HH:mm:ss.fff";
-        
+
```

Comment:
nit: revert space

Reply:
I'm applying the same logic here: I removed white space but the effect/appearance is the same. There is no need for space characters on an empty line.